odoo_tools/utils/pending_merge.py

```diff
@@ -236,6 +283,30 @@ def add_pending_pull_request(self, upstream, pull_id):
         self.update_merges_config(conf)
         return True
 
+    def _get_pending_commit_lines(self, upstream: str, commit_sha: str):
+        """Return the lines to add to the merges file for a given commit."""
+        if self.template_version == "1":
+            return [
+                f'git fetch {upstream} {commit_sha}',
+                f'git am "$(git format-patch -1 {commit_sha} -o ../patches)"',
+            ]
+        # self.template_version == "2"
+        if self.name.lower() in ("odoo", "enterprise"):
+            # For Odoo repositories, we store the patch in a subdirectory
+            # This patch will be applied when the image is built.
+            patch_dir = f"../../patches/{self.name}"
+            return [
+                f'git fetch {upstream} {commit_sha}',
+                f'git am "$(git format-patch -1 {commit_sha} -o {patch_dir})"',
+            ]
+        else:
+            # For external repositories, we just cherry-pick the commit,
+            # as we don't need to store the patch file.
```

Comment:
Is my assumption correct?

Reply:
it is (but  boy, getting back into this project is painful...)

Reply:
yes. 

For external repositories we need to cherry pick, and then at some point to push the commit to a consolidation branch, and then to get the requirements.txt file to install from that branch.